misk-redis/src/test/kotlin/misk/redis/RedisAuthPasswordEnvTest.kt

```diff
@@ -0,0 +1,53 @@
+package misk.redis
+
+import com.google.inject.Guice.createInjector
+import com.google.inject.ProvisionException
+import misk.MiskTestingServiceModule
+import misk.environment.DeploymentModule
+import misk.inject.KAbstractModule
+import misk.inject.keyOf
+import misk.redis.testing.DockerRedis
+import misk.testing.MiskExternalDependency
+import misk.testing.MiskTest
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import redis.clients.jedis.JedisPoolConfig
+import wisp.deployment.PRODUCTION
+import wisp.deployment.TESTING
+import javax.inject.Inject
+
+@MiskTest
+class RedisAuthPasswordEnvTest {
+  @Suppress("unused")
+  @MiskExternalDependency
+  private val dockerRedis = DockerRedis // Start Redis so that we can properly provision a client.
+
+  @Test fun `injection succeeds with password-less config in fake environments`() {
+    assertThat(DockerRedis.config.values.first().redis_auth_password).isEmpty()
+    val injector = createInjector(fakeEnv, realRedisModule)
+    val redis = injector.getInstance(keyOf<RedisConsumer>()).redis
+    assertThat(redis).isInstanceOf(RealRedis::class.java)
```

Comment:
should we add some operation to validate we can write/read to docker redis?

Reply:
Done!